projects/make-a-vis/src/lib/shared/services/legend/legend.service.ts

```diff
@@ -0,0 +1,63 @@
+import { Injectable } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import { DefaultGraphicSymbol, GraphicSymbol, GraphicVariable, ProjectSerializerService, Project, Visualization } from '@dvl-fw/core';
+import { Subject } from 'rxjs';
+import { uniqueId } from 'lodash';
+
+import { getLoadedProjectSelector, SidenavState } from '../../../toolbar/shared/store';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LegendService {
+
+  constructor(private store: Store<SidenavState>, private serializer: ProjectSerializerService) {}
+
+  public createLegend(template, gv, gs): Subject<Visualization> {
```

Comment:
Add type information to method and return an Observable (not good practice to return a Subject. can use subject.asObservable() to get this).